bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtocol.java

```diff
@@ -562,6 +562,10 @@ class BatchedReadResponse extends Response implements ReferenceCounted {
         final long requestId;
         final ByteBufList data;
 
+        BatchedReadResponse(byte protocolVersion, int errorCode, long ledgerId, long entryId, long requestId) {
+            this(protocolVersion, errorCode, ledgerId, entryId, requestId, ByteBufList.get());
```

Comment:
Reminder: No matter what the error code is, don't forget to release the ByteBufList in the client side.

Reply:
Sure, the error BatchedReadResponse will also be released.